contracts/staking/ServiceStakingBase.sol

```diff
@@ -505,30 +584,43 @@ abstract contract ServiceStakingBase is ERC721TokenReceiver, IErrorsRegistries {
 
         // If service Ids are returned, then the checkpoint takes place
         if (serviceIds.length > 0) {
-            // Updated current service nonces
+            // Record service inactivities and updated current service nonces
             for (uint256 i = 0; i < serviceIds.length; ++i) {
                 // Get the current service Id
-                uint256 curServiceId = serviceIds[i];
+                curServiceId = serviceIds[i];
+                // Record service nonces
                 mapServiceInfo[curServiceId].nonces = serviceNonces[i];
 
                 // Increase service inactivity if it is greater than zero
                 if (serviceInactivity[i] > 0) {
-                    mapServiceInfo[curServiceId].inactivity += serviceInactivity[i];
+                    // Get the overall continuous service inactivity
+                    uint256 inactivity = mapServiceInfo[curServiceId].inactivity + serviceInactivity[i];
+                    mapServiceInfo[curServiceId].inactivity = inactivity;
+                    // Check for the maximum allowed inactivity time
+                    if (inactivity > maxAllowedInactivity) {
+                        // Evict a service if it has been inactive for more than a maximum allowed inactivity time
+                        evictServiceIds[i] = curServiceId;
+                    }
```

Comment:
If inactivity is bigger than the allowed one, the service assigned to evicted services list.